### ISSUE TITLE
Grid Editor throws IllegalStateException

### ISSUE THREAD

**Issue by KochTobi** [vaadin/flow#20370]
### Description of the bug

When having a grid displaying java records, from 24.5.x on the `editItem` method throws an exception. This did not occur before as there is a custom renderer and it was not an issue before 24.5.x

Seems to have something to do on how the Binder for the editor is initialized.
```java
var binder = new Binder<>(MyClass.class);
```
versus
```java
Binder<MyClass> binder = new Binder<>();
```

### Expected behavior

The editor component should show loaded with the information from the record (as was the case in 24.4.13) no matter how the binder is initialized.

### Minimal reproducible example

https://github.com/KochTobi/broken-vaadin-binder
You can find a working example at
https://github.com/KochTobi/broken-vaadin-binder/tree/working-example

### Versions

- Vaadin / Flow version: 24.5.2
- Java version: 21
- OS version: MacOs 14.7
- Browser version (if applicable): Firefox 131.0.3
- Application Server (if applicable):
- IDE (if applicable):

**Comment by mcollovati** [comment]
Java records are immutable, so incompatible with the semantic of `setBean()`.
Here's the link to related [documentation](https://vaadin.com/docs/latest/flow/binding-data/components-binder-load#using-java-records-with-binder)

**Comment by KochTobi** [comment]
Yes but the Editor in question is provided by vaadin itself.

**Comment by mcollovati** [comment]
Yes, sorry. I missed that piece. I'll transfer the issue to the flow-components repository
